gbus/tx/mysql/timeout.go

```diff
@@ -0,0 +1,223 @@
+package mysql
+
+import (
+	"database/sql"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/wework/grabbit/gbus"
+)
+
+var _ gbus.TimeoutManager = &TimeoutManager{}
+
+//TimeoutManager is a mysql implementation of a persistent timeoutmanager
+type TimeoutManager struct {
+	Bus               gbus.Bus
+	Log               func() logrus.FieldLogger
+	TimeoutSaga       func(*sql.Tx, string) error
+	Txp               gbus.TxProvider
+	SvcName           string
+	timeoutsTableName string
+	exit              chan bool
+}
+
+func (tm *TimeoutManager) ensureSchema() error {
+	tblName := tm.timeoutsTableName
+	tx, e := tm.Txp.New()
+	if e != nil {
+		tm.Log().WithError(e).Error("failed to create schema for mysql timeout manager")
+		return e
+	}
+
+	createTableSQL := `CREATE TABLE IF NOT EXISTS ` + tblName + ` (
+      rec_id INT PRIMARY KEY AUTO_INCREMENT,
+      saga_id VARCHAR(255) UNIQUE NOT NULL,
+	  timeout DATETIME NOT NULL,
+	  INDEX (timeout),
+	  INDEX (saga_id)
+	 )`
+
+	if _, e := tx.Exec(createTableSQL); e != nil {
+		if rbkErr := tx.Rollback(); rbkErr != nil {
+			tm.Log().Warn("timeout manager failed to rollback transaction")
+		}
+		return e
+	}
+	return tx.Commit()
+}
+
+func (tm *TimeoutManager) purge() error {
+	purgeSQL := `DELETE FROM ` + tm.timeoutsTableName
```

Comment:
This can be `truncate` since you probably did this for tests where the I'd of the primary key might have impact...

Reply:
@vladshub  truncate gives me a problem when running in tests....I am assuming purge options will be used only in local development and not in production so I do not see why this may be a problem